File: python/convert-an-array-into-a-2d-array-with-conditions.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        output = []
        while nums:
            current = nums[-1]
            i = 0
            while len(nums) and nums[-1] == current:
                if i == len(output):
                    output.append([])
                output[i].append(nums.pop())
                i += 1
        return output

File: rust/convert-an-array-into-a-2d-array-with-conditions.rs
impl Solution {
    pub fn find_matrix(mut nums: Vec<i32>) -> Vec<Vec<i32>> {
        nums.sort();
        let mut output = Vec::new();
        let mut j = 0;
        for i in 0..nums.len() {
            if i == 0 || nums[i - 1] != nums[i] {
                j = 0;
            }
            if j == output.len() {
                output.push(Vec::new());
            }
            output[j].push(nums[i]);
            j += 1;
        }
        return output;
    }
}
